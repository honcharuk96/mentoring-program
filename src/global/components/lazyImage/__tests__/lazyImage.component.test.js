import React from 'react';
import { render } from '@testing-library/react';
import { LazyImage } from '../lazyImage.compoent';

const observe = jest.fn();
const unobserve = jest.fn();

window.IntersectionObserver = jest.fn(() => ({
  observe,
  unobserve,
}));

describe('<LazyImage />', () => {
  it('should render LazyImage isBanner', () => {
    const { asFragment } = render(<LazyImage alt={'test'} src={'https:/test.com'} isBanner={true} />);
    // expect(observe).toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render LazyImage', () => {
    const { asFragment } = render(<LazyImage alt={'test'} src={'https:/test.com'} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
