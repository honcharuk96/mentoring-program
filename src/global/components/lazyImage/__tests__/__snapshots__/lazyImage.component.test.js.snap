// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LazyImage /> should render LazyImage 1`] = `
<DocumentFragment>
  <img
    alt="test"
    class="lazyImagecompoent__Image-sc-inbsub-0 cvpBYK"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkqAcAAIUAgUW0RjgAAAAASUVORK5CYII="
  />
</DocumentFragment>
`;

exports[`<LazyImage /> should render LazyImage isBanner 1`] = `
<DocumentFragment>
  <img
    alt="test"
    class="lazyImagecompoent__Image-sc-inbsub-0 gaLdew"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkqAcAAIUAgUW0RjgAAAAASUVORK5CYII="
  />
</DocumentFragment>
`;
